moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -153,91 +163,96 @@ void JogCalcs::startMainLoop(JogArmShared& shared_variables, std::mutex& mutex)
-    bool stale_command = shared_variables.command_is_stale;
-    mutex.unlock();
-
-    if (stale_command || (zero_cartesian_cmd_flag && zero_joint_cmd_flag))
-    {
-      suddenHalt(outgoing_command_);
-      zero_cartesian_cmd_flag = true;
-      zero_joint_cmd_flag = true;
-    }
-
-    bool valid_nonzero_command = !zero_cartesian_cmd_flag || !zero_joint_cmd_flag;
-
-    // Send the newest target joints
-    if (!outgoing_command_.joint_names.empty())
-    {
-      mutex.lock();
-      // If everything normal, share the new traj to be published
-      if (valid_nonzero_command)
+      // Prioritize cartesian jogging above joint jogging
+      if (!zero_cartesian_cmd_flag)
       {
-        shared_variables.outgoing_command = outgoing_command_;
-        shared_variables.ok_to_publish = true;
+        mutex.lock();
+        cartesian_deltas = shared_variables.command_deltas;
+        mutex.unlock();
+
+        if (!cartesianJogCalcs(cartesian_deltas, shared_variables, mutex))
+          continue;
       }
-      // Skip the jogging publication if all inputs have been zero for several cycles in a row.
-      // num_outgoing_halt_msgs_to_publish == 0 signifies that we should keep republishing forever.
-      else if ((parameters_.num_outgoing_halt_msgs_to_publish != 0) &&
-               (zero_velocity_count > parameters_.num_outgoing_halt_msgs_to_publish))
+      else if (!zero_joint_cmd_flag)
       {
-        shared_variables.ok_to_publish = false;
-        // Reset the velocity filters so robot doesn't jump when commands resume
-        resetVelocityFilters();
+        mutex.lock();
+        joint_deltas = shared_variables.joint_command_deltas;
+        mutex.unlock();
+
+        if (!jointJogCalcs(joint_deltas, shared_variables))
+          continue;
       }
-      // The command is invalid but we are publishing num_outgoing_halt_msgs_to_publish
       else
       {
-        shared_variables.outgoing_command = outgoing_command_;
-        shared_variables.ok_to_publish = true;
+        outgoing_command_ = composeJointTrajMessage(joint_state_);
       }
+
+      // Halt if the command is stale or inputs are all zero, or commands were zero
+      mutex.lock();
+      bool stale_command = shared_variables.command_is_stale;
       mutex.unlock();
 
-      // Store last zero-velocity message flag to prevent superfluous warnings.
-      // Cartesian and joint commands must both be zero.
-      if (zero_cartesian_cmd_flag && zero_joint_cmd_flag)
+      if (stale_command || (zero_cartesian_cmd_flag && zero_joint_cmd_flag))
       {
-        // Avoid overflow
-        if (zero_velocity_count < INT_MAX)
-          ++zero_velocity_count;
+        suddenHalt(outgoing_command_);
+        zero_cartesian_cmd_flag = true;
+        zero_joint_cmd_flag = true;
+      }
+
+      bool valid_nonzero_command = !zero_cartesian_cmd_flag || !zero_joint_cmd_flag;
+
+      // Send the newest target joints
+      if (!outgoing_command_.joint_names.empty())
```

Comment:
Are there ever no joint names specified in the message at this point?

Reply:
not that i know of, will try deleting it

Reply:
deleting it _seems_ fine...